File: prelim_analysis/meta_merge.R

# BD
abmap$Bd_status <- 0
for (i in seq_len(nrow(mgmap))) {
         if (mgmap$Bd_status[i] == "Negative") {
             mgmap$Bd_status[i] <- 0
         } else {
             mgmap$Bd_status[i] <- 1
         }
}
mgmap$Bd_status <- as.numeric(mgmap$Bd_status)
for (i in which(!is.na(semap$Bd_status) == TRUE)) {
         if (semap$Bd_status[i] == "Negative") {
             semap$Bd_status[i] <- 0
         } else if (semap$Bd_status[i] == "Positive") {
             semap$Bd_status[i] <- 1
         }
}
semap$Bd_status <- as.numeric(semap$Bd_status) # }}}

## {{{ DIVIDE CALIFORNIA INTO FOUR REGIONS
abmap$State_Region <- 0
for (i in seq_len(nrow(abmap))) {
  if (abmap$Site[i] == "Alameda" || abmap$Site[i] == "Monterey") {
    abmap$State_Region[i] <- "Coastal California"
  } else if (abmap$Site[i] == "Jackson_State_Forest" ||
             abmap$Site[i] == "Siskiyou" || abmap$Site == "Shasta" ||
             abmap$Site[i] == "Humboldt" ||
             abmap$Site[i] == "Leggett") {
    abmap$State_Region[i] <- "Northern California"
  } else if (abmap$Site[i] == "Sierra_National_Forest") {
    abmap$State_Region[i] <- "Sierra Nevada"
  } else {
    abmap$State_Region[i] <- "Southern California"
  }
}
spimap$State_Region <- 0
for (i in seq_len(nrow(spimap))) {
  if (spimap$Site[i] == "Alameda" || spimap$Site[i] ==
      "Santa.Cruz" || spimap$pop[i] == "north.bay") {
    spimap$State_Region[i] <- "Coastal California"
  } else if (spimap$Site[i] == "Madera") {
    spimap$State_Region[i] <- "Sierra Nevada"
  } else if (spimap$pop[i] == "annadel") {
    spimap$State_Region[i] <- "Northern California"
  } else {
    spimap$State_Region[i] <- "Southern California"
  }
}
semap$State_Region <- "Sierra Nevada" # }}}

## {{{ GPS & COUNTY DATA
for (i in seq_len(nrow(semap))) {
    if (i < 32) {
        semap$Latitude[i] <- 38.934
        semap$Longitude[i] <- -120.1475
    } else if (32 < i & i < 64) {
        semap$Latitude[i] <- 38.38841
        semap$Longitude[i] <- -120.1595
    } else if (64 < i & i < 96) {
        semap$Latitude[i] <- 38.402
        semap$Longitude[i] <- -120.1602
    } else {
        semap$Latitude[i] <- 39.008395
        semap$Longitude[i] <- -120.21257
    }
}

#County
library(maps)
semap$Site <- map.where(database = "county",
                    semap$Longitude, semap$Latitude)
abmap$Site <- map.where(database = "county",
                    abmap$Longitude, abmap$Latitude)
semap$Site <- sub("...........", "", semap$Site)
abmap$Site <- sub("...........", "", abmap$Site) # }}}

## {{{ JOIN & EXPORT TABLES ------
meta_1 <- full_join(semap, mgmap, by = c("SampleID", "Bd_status",
                                         "State_Region", "Site", "Dataset",
                                         "Longitude", "Latitude",
                                         "Family", "Genus", "Species",
                                         "Order"), copy = TRUE)
meta_2 <- full_join(abmap, spimap, by = c("SampleID", "Genus", "Species",
                                          "Family", "Order", "Site", "Dataset",
                                          "Latitude", "Longitude", "subspecies",
                                           "State_Region"), copy = TRUE)
meta_3 <- full_join(meta_1, meta_2, by = c("SampleID", "Genus", "Species",
                                           "Family", "Order", "subspecies",
                                           "Habitat", "State_Region", "Dataset",
                                           "Latitude",
                                           "Longitude", "Site", "Bd_status"),
                    copy = TRUE)
write.table(meta_3, file = "merged_metadata.txt", append = FALSE, sep = "\t",
            row.names = FALSE, quote = FALSE, col.names = TRUE) #}}}

File: prelim_analysis/meta_merge.py
## {{{ META MERGE: Merge mapping files from each dataset
import pandas as pd
import numpy as np
abmap = pd.read_table("~/Documents/amphibian_meta_project/meta_analysis/" +
        "qiime_analyses/AB_mapping_file.txt", sep='\t')
mgmap = pd.read_table('~/Documents/amphibian_meta_project/meta_analysis/' +
        'qiime_analyses/MG_Mapping_File_GUA_with_Bd.txt', sep='\t')
spimap = pd.read_table('~/Documents/amphibian_meta_project/meta_analysis/' +
        'qiime_analyses/SPI_mapping_file.txt', sep='\t')
semap = pd.read_table('~/Documents/amphibian_meta_project/meta_analysis/' +
        'qiime_analyses/SE_mapping_file.txt', sep='\t') # }}}

## {{{ DIVIDE CALIFORNIA INTO REGIONS
abmap['State_Region'] = np.nan
abmap.loc[(abmap['Site']=='Alameda') | (abmap['Site']=='Monterey'),
    'State_Region'] = 'Coastal California'
abmap.loc[(abmap['Site']=='Jackson_State_Forest') |
        (abmap['Site']=='Siskiyou') | (abmap['Site']=='Shasta') |
        (abmap['Site']=='Humboldt') | (abmap['Site']=='Leggett'),
        'State_Region'] = 'Northern California'
abmap.loc[(abmap['Site']=='Sierra_National_Forest'),
        'State_Region'] = 'Sierra Lowland'
abmap['State_Region'] = abmap['State_Region'].replace(
        np.nan, 'Southern California')

spimap['State_Region'] = np.nan
spimap.loc[(spimap['Site']=='Alameda') | (spimap['Site']=='Santa.Cruz') |
        (spimap['pop']=='north.bay'), 'State_Region'] = 'Coastal California'
spimap.loc[(spimap['Site']=='Madera'), 'State_Region'] = 'Sierra Lowland'
spimap.loc[(spimap['pop']=='annadel'), 'State_Region'] = 'Northern California'
spimap['State_Region'] = spimap['State_Region'].replace(np.nan,
        'Southern California')
spimap['State_Region'].value_counts()

semap['State_Region'] = 'Sierra Highland'
semap['Latitude'] = np.nan
semap['Longitude'] = np.nan
semap.loc[(semap['site_id']=='Pyramid_Valley'), 'Latitude'] = 38.825959
semap.loc[(semap['site_id']=='Rivendell'), 'Latitude'] = 38.848118
semap.loc[(semap['site_id']=='Pyramid_Valley'), 'Longitude'] = -120.142295
semap.loc[(semap['site_id']=='Rivendell'), 'Longitude'] = -120.163238 # }}}

## {{{ FORMAT COLUMNS
abmap = abmap.drop(['Higher_Clade', 'Lower_Clade', 'Habitat_Type',
    'Bd_status'], axis=1)
mgmap = mgmap.drop(['BarcodeSequence', 'LinkerPrimerSequence', 'ReversePrimer',
    'Collector_number', 'subspecies', 'Country', 'County_Municipio', 'Site',
    'Date', 'Sex', 'Age_Class', 'ZE', 'Log_ZE', 'Elevation', 'Mountain_range',
    'Geology', 'Habitat', 'Description', 'Bd_Status'], axis=1)
spimap = spimap.drop(['BarcodeSequence', 'LinkerPrimerSequence',
    'ReversePrimer', 'type', 'project1', 'project2', 'subspecies', 'pop', 'age',
    'sex', 'cov.obj', 'svl', 'tot.lgth', 'wgt', 'soil.moist', 'soil.temp',
    'cov.wd', 'cov.lth', 'date', 'Description'], axis=1)
semap = semap.drop(['BarcodeSequence', 'LinkerPrimerSequence', 'ReversePrimer',
    'Sample_Name', 'frog_id', 'lifestage', 'gosner_stage', 'month', 'year',
    'ZE', 'survey_date', 'pit_tag_id', 'frog_sex', 'extraction_date',
    'frog_location', 'frog_weight', 'frog_svl', 'swabber_name', 'Description',
    'Bd_status'],
    axis=1)

abmap = abmap.rename(columns={'Sample_Type':'Order'})
semap = semap.rename(columns={'site_id':'Site'})
mgmap = mgmap.rename(columns={'State_Province': 'Site'})
spimap = spimap.rename(columns={'county': 'Site',
    'lat':'Latitude', 'long':'Longitude'})
mgmap = mgmap.drop([0, 78, 79, 80, 81, 82])
mgmap['Latitude'] = mgmap['Latitude'].astype('float')
mgmap['Longitude'] = mgmap['Longitude'].astype('float')
mgmap['State_Region'] = "Central America"
abmap['Dataset'] = "Bird et al., 2018"
mgmap['Dataset'] = "Ellison et al., 2018"
semap['Dataset'] = "Ellison et al., 2019"
spimap['Dataset'] = "Prado-Irwin et al., 2017" # }}}

## {{{ TAXONOMY & BD
abmap['Family'] = "Plethodontidae"
abmap['Order'] = "Salamander"
spimap['Family'] = "Plethodontidae"
spimap['Order'] = "Salamander"
spimap['Genus'] = "Ensatina"
spimap['Species'] = "Ensatina_eschscholtzii"
semap['Species'] = "Rana_sierrae"
semap['Genus'] = "Rana"
semap['Family'] = "Ranidae"
semap['Order'] = "Frog"
mgmap['Order'] = mgmap['Order'].apply(lambda x: 'Salamander' if x=='Caudata'
        else 'Frog') # }}}

## {{{ JOIN & EXPORT TABLES ------
dfs = [abmap, spimap, mgmap, semap]

from functools import reduce
mm = reduce(lambda left,right: pd.merge(left,right, on=["SampleID",
    "State_Region", "Dataset", "Longitude", "Latitude", "Family", "Genus",
    "Species", "Order", "Site"], how='outer'), dfs)

mm.to_csv('~/Documents/amphibian_meta_project/meta_analysis/' +
        'qiime_analyses/merged_metadata.txt', sep="\t", header=True,
        index=False)
# }}}
